homeassistant/components/backup/manager.py

```diff
@@ -262,80 +279,48 @@ async def async_upload_backup(self, *, slug: str, **kwargs: Any) -> None:
                         protected=backup.protected,
                     ),
                 )
-                for agent in self.backup_agents.values()
+                for agent_id in agent_ids
             ),
             return_exceptions=True,
         )
         for result in sync_backup_results:
             if isinstance(result, Exception):
                 LOGGER.error("Error during backup upload - %s", result)
+        # TODO: Reset self.syncing in a finally block
         self.syncing = False
 
-    async def load_backups(self) -> None:
-        """Load data of stored backup files."""
-        backups = await self.hass.async_add_executor_job(self._read_backups)
-        LOGGER.debug("Loaded %s local backups", len(backups))
-        self.backups = backups
-        self.loaded_backups = True
-
-    def _read_backups(self) -> dict[str, Backup]:
-        """Read backups from disk."""
-        backups: dict[str, Backup] = {}
-        for backup_path in self.backup_dir.glob("*.tar"):
-            try:
-                with tarfile.open(backup_path, "r:", bufsize=BUF_SIZE) as backup_file:
-                    if data_file := backup_file.extractfile("./backup.json"):
-                        data = json_loads_object(data_file.read())
-                        backup = Backup(
-                            slug=cast(str, data["slug"]),
-                            name=cast(str, data["name"]),
-                            date=cast(str, data["date"]),
-                            path=backup_path,
-                            size=round(backup_path.stat().st_size / 1_048_576, 2),
-                            protected=cast(bool, data.get("protected", False)),
-                        )
-                        backups[backup.slug] = backup
-            except (OSError, TarError, json.JSONDecodeError, KeyError) as err:
-                LOGGER.warning("Unable to read backup %s: %s", backup_path, err)
-        return backups
-
     async def async_get_backups(self, **kwargs: Any) -> dict[str, Backup]:
         """Return backups."""
```

Comment:
I think it makes sense the manager knows about all backups across all agents.
The manager should not have to ask each time though, there needs to be some caching. Also, either frontend or an agent should be able to initiate a refresh.

Reply:
The caching needs to happen in the individual agents, they know what's reasonable, not the manager